pkg/middleware/auth.go

```diff
@@ -146,6 +148,52 @@ func Auth(options *AuthOptions) web.Handler {
 	}
 }
 
+func RoleAppPluginAuthAndSignedIn(ps pluginstore.Store) web.Handler {
+	return func(c *contextmodel.ReqContext) {
+		if !c.IsSignedIn {
+			notAuthorized(c)
+			return
+		}
+
+		pluginID := web.Params(c.Req)[":id"]
+		p, exists := ps.Plugin(c.Req.Context(), pluginID)
+		if !exists {
+			c.JsonApiErr(http.StatusNotFound, "Plugin not found", nil)
+			return
+		}
+
+		normalizePath := func(p string) string {
+			return filepath.Clean(p)
+		}
+
+		//found := false
+		allowed := false
+		path := normalizePath(c.Req.RequestURI)
+		for _, i := range p.Includes {
+			if i.Type != "page" {
+				continue
+			}
+
+			if normalizePath(i.Path) == path {
+				//found = true
+				if i.Role == "" || c.HasRole(i.Role) {
+					allowed = true
+				}
+				break
+			}
+		}
+
+		//if !found {
```

Comment:
TODO: Awaiting confirmation to see if we need to have separate behaviour for this or not